pkg/identity/service.go

```diff
@@ -73,10 +76,34 @@ func (svc *Service) Initialize() error {
 	if err != nil {
 		return err
 	}
+
+	cfg, err := config.Global(consts.MountPathGlobalConfig)
+	if err != nil {
+		return err
+	}
+
+	if cfg.GetTlsManager() != consts.TLSManagerInternal {
+		if newAnchorsPEM, err := issuercerts.LoadTrustAnchorsFromFile(svc.anchorsPathCrt); err == nil {
+			newTrustAnchors, err := tls.DecodePEMCertPool(newAnchorsPEM)
+			if err != nil {
+				return err
+			}
+			svc.updateAnchors(newTrustAnchors)
+		} else {
+			return err
+		}
+	}
 	svc.updateIssuer(credentials)
 	return nil
 }
 
+func (svc *Service) updateAnchors(newAnchors *x509.CertPool) {
+	svc.issuerMutex.Lock()
+	svc.trustAnchors = newAnchors
+	log.Debug("Trust anchors have been updated")
+	svc.issuerMutex.Unlock()
+}
+
```

Comment:
Have you tested this codepath and ensured nothing actually breaks. When that anchor gets updated, it seems nothing takes care to update the anchors that are embedded in the injected proxies. next time the proxy fires a CSR, it will fail to get a new cert, which will most likely break things in a pretty bad way.

Reply:
Hi @zaharidichev! 

You're correct. I don't have anything yet to update the injected proxies with the new anchor. I think they receive the anchor values from env vars at the time of injection. Is there a safer way to restart all the injected proxies without any downtime?

Should the proxies watch the issuer secret to pull the trust anchor instead? We could then have them be updated without any downtime like we do for identity and I recently added for the webhooks.

Thoughts?

Reply:
So yes, having the proxies pick up the anchors is a pretty essential part of all this. The problem with them watching the fs for the secret being updated is that you will effectivelly need to mount that secret in pretty much every pod that has a proxy injected. I am not sure this is desirable. I have to think about that a bit more.

Reply:
As updates are not guaranteed, you'll always loose traffic which is why we've not done this for the proxies so far.